proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,204 @@
+syntax = "proto3";
+
+package com.mintter.groups.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "mintter/backend/genproto/groups/v1alpha;groups";
+
+// Groups service exposes the functionality for managing group entities.
+// This API uses fully-qualified EID as group IDs in all the requests.
+service Groups {
+  // Creates a new group.
+  rpc CreateGroup(CreateGroupRequest) returns (Group);
+
+  // Gets a group by ID.
+  rpc GetGroup(GetGroupRequest) returns (Group);
+
+  // Updates a group.
+  rpc UpdateGroup(UpdateGroupRequest) returns (Group);
+
+  // Lists members of a group.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Lists content of a group.
+  rpc ListContent(ListContentRequest) returns (ListContentResponse);
+
+  // Lists groups.
+  rpc ListGroups(ListGroupsRequest) returns (ListGroupsResponse);
+}
+
+// Request to create a group.
+message CreateGroupRequest {
+  // Required. Title of the new group.
+  string title = 1;
+
+  // Optional. Description of the new group.
+  string description = 2;
+
+  // Optional. List of initial members for the new group.
+  // Members can also be managed with separate requests after group is already created.
+  map<string, Role> members = 3;
+}
+
+// Request to get a group.
+message GetGroupRequest {
+  // Required. ID of the group to get.
+  // Must be a fully-qualified EID.
+  string id = 1;
+
+  // Optional. Version of the group to get information about.
+  // If empty, latest_version is will be used.
+  string version = 2;
+}
+
+// Request to update a group.
+// Some fields are required, even if unchanged,
+// to distinguish between setting the value to an empty string, from an unchanged value.
+// Read the comments!
+message UpdateGroupRequest {
+  // ID of the group to update.
+  string id = 1;
+
+  // Required. Title of the Group.
+  // Must always be provided, even if unchanged.
+  string title = 2;
+
+  // Required. Description of the Group. Can be empty string.
+  // Must always be provided, even if unchanged.
+  string description = 3;
+
+  // Optional. List of members to be updated in the Group.
+  // Key is the member's Account ID,
+  // value is the Role.
+  // To remove a member from the group, set the role to unspecified.
+  map<string, Role> updated_members = 4;
+
+  // Optional. List of content to be updated in the Group.
+  // Key is a pretty path on which the content is published,
+  // value is a Hyperdocs URL of the content.
+  // To unpublish content set the pretty path to an empty string.
+  map<string, string> updated_content = 5;
+}
+
+// Response to list members.
+message ListMembersRequest {
+  // Required. ID of the group to list members of.
+  string id = 1;
+
+  // Optional. Version of the group to list members of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
```

Comment:
I know it's an implementation detail but I wonder how are you going to manage versions in member lists. I don't see the logic needed in sqlite only.

Reply:
Yeah, that's some tricky thing to do seem like. I have some rough ideas, but I guess this will be uncovered during the implementation.

I'll probably have to store added and removed members in a separate sqlite table and then "fold" these records up to the requested version to get the list of members at that version.